iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -98,6 +102,9 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
             {
                 using var ctb = new CancellationTokenBundle(_operationTimeout, cancellationToken);
                 await _amqpUnit.OpenAsync(ctb.Token).ConfigureAwait(false);
+
+                // Create a timer to remove any expired messages. The timer would invoke callback after first hour of execution and then after every minute.
```

Comment:
Checking for expired messages every minute seems a bit aggressive. Why not check once per hour? We don't need this map to be up-to-date so much as we need it to just not grow over time.